lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1040,373 @@ spec = do
+
+        -- Incoming tx inputs and outpus size
+        let outsIn = getFromResponseList 1 outputs r
+        let insIn = getFromResponseList 1 inputs r
+        length (outsIn :: [AddressAmount t]) `shouldBe` 1000
+        length (insIn :: [ApiTxInput t]) `shouldBe` 1
+
+        verify r
+            [ expectListSizeEqual 2
+            , expectListItemFieldEqual 0 direction Outgoing
+            , expectListItemFieldBetween 0 amount (feeMin + 1, feeMax + 1)
+            , expectListItemFieldEqual 0 status InLedger
+            , expectListItemFieldEqual 1 direction Incoming
+            , expectListItemFieldEqual 1 amount 1_000_000_000_000
+            , expectListItemFieldEqual 1 status InLedger
+            ]
+
+    -- This scenario covers the following matrix of cases. Cases were generated
+    -- using one of pairwise test cases generation tools available online.
+    -- +----+----------+----------+------------+--------------+
+    --     |  start   |   end    |   order    |    result    |
+    -- +----+----------+----------+------------+--------------+
+    --   1 | edge     | edge     | ascending  | 2 ascending  |
+    --   2 | edge     | edge + 1 | descending | 2 descending |
+    --   3 | edge     | edge - 1 | empty      | 1st one      |
+    --   4 | edge     | empty    | empty      | 2 descending |
+    --   5 | edge + 1 | edge + 1 | empty      | 2nd one      |
+    --   6 | edge + 1 | edge - 1 | empty      | none         |
+    --   7 | edge + 1 | empty    | ascending  | 2nd one      |
+    --   8 | edge + 1 | edge     | descending | 2nd one      |
+    --   9 | edge - 1 | edge - 1 | ascending  | 1st one      |
+    --  10 | edge - 1 | empty    | descending | 2 descending |
+    --  11 | edge - 1 | edge     | empty      | 2 descending |
+    --  12 | edge - 1 | edge + 1 | empty      | 2 descending |
+    --  13 | empty    | empty    | empty      | 2 descending |
+    --  14 | empty    | edge     | empty      | 2 descending |
+    --  15 | empty    | edge + 1 | ascending  | 2 ascending  |
+    --  16 | empty    | edge - 1 | descending | 1st one      |
+    --  17 | t1       | t1       | empty      | 1st one      |
+    --  18 | t2       | t2       | descending | 2nd one      |
+    -- +----+----------+----------+------------+--------------+
+    it "TRANS_LIST_02,03 - Can limit/order results with start, end and order"
+        $ \ctx -> do
+        (_, w, [(t1, a1), (t2, a2)]) <- fixtureWalletManyTxs ctx [10,20]
+        let matrix =
+                [
+                -- 1
+                  ( toQueryString
+                        [ ( "start", utcIso8601ToText t1)
+                        , ( "end", utcIso8601ToText t2 )
+                        , ( "order", "ascending" )
+                        ]
+                  ,  [ expectListSizeEqual 2
+                     , expectListItemFieldEqual 0 amount a1
+                     , expectListItemFieldEqual 1 amount a2
+                     ]
+                  )
```

Comment:
For better readability, I'd recommend to switch from a tuple to a record type and used the constructor with the named fields to make it easy for the reader to go through the matrix :+1:
Tuples are the poor man's sum types, acceptable for short arguments. Here, since we have many cases with rather long arguments, a record would combine them in a much more readable way!